### PR TITLE
base image with propper cmake

### PR DESCRIPTION
**Pull request by FG-TUM**
# Description

Add a base image for all autopas containers with the dependencies that are shared by most. Advantage: If we want to upgrade e.g. cmake again we only need to do it there and then just rebuild all images.

- [x] base image
- [x] build cmake from source
- [x] use base image in all other images
- [x] CMD to make containers more easy to use

### CODE REVIEW

File: buildenv/clang/Dockerfile [hunk 1]
```diff
@@ -38,9 +19,15 @@ RUN true \
...
 	&& rm -rf /var/lib/apt/lists/*
+RUN pip3 install --upgrade cmake-format
 
 # generate links to clang
 RUN ln -s /usr/bin/clang-${CLANGVERSION} /usr/bin/clang
 RUN ln -s /usr/bin/clang++-${CLANGVERSION} /usr/bin/clang++
 RUN ln -s /usr/bin/llvm-config-${CLANGVERSION} /usr/bin/llvm-config
+
+
+CMD ["sh", "-c", "echo 'Formatting code...' && cmake -G Ninja -DAUTOPAS_OPENMP=ON ../.. > /dev/null && ninja clangformat > /dev/null && ninja cmakeformat > /dev/null && echo 'done' && ninja && ctest"]
```
Comment (SteffenSeckler) [inline buildenv/clang/Dockerfile]:
```suggestion
CMD ["sh", "-c", "echo 'Formatting code...' && cmake -G Ninja -DAUTOPAS_OPENMP=ON ../.. > /dev/null && ninja clangformat > /dev/null && ninja cmakeformat > /dev/null && echo 'done' && ninja && ninja test"]
```

File: buildenv/doxygen/Dockerfile [hunk 2]
```diff
@@ -1,13 +1,11 @@
...
-		build-essential \
-		python-pip \
 		doxygen \
 		graphviz \
 	&& rm -rf /var/lib/apt/lists/*
-# install current cmake
-RUN pip install --upgrade pip
-RUN pip install --upgrade cmake
+
+CMD ["sh", "-c", "cmake ../.. >/dev/null && echo \"\ndoxygen warnings:\" && make doc_doxygen > /dev/null"]
```
Comment (SteffenSeckler) [inline buildenv/doxygen/Dockerfile:11]:
```suggestion
CMD ["sh", "-c", "cmake -GNinja ../.. >/dev/null && echo \"\ndoxygen warnings:\" && ninja doc_doxygen > /dev/null"]
```

**Review by SteffenSeckler — APPROVED** [review]

File: buildenv/base/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		ninja-build \
+		wget \
+		gdb \
+		git \
+		ccache \
+		ssh \
+		gnupg \
+	&& rm -rf /var/lib/apt/lists/*
+
+# we might want to fix the certificate stuff
```
Comment (FG-TUM) [inline buildenv/base/Dockerfile:17]: did you look into this?

File: buildenv/doxygen/Dockerfile [hunk 4]
```diff
@@ -1,13 +1,11 @@
-FROM ubuntu:16.04
+FROM autopas/build-base:latest
```
Comment (FG-TUM) [inline buildenv/doxygen/Dockerfile:1]: this and the gcc one are the only ones where you added the `latest` tag

**Pull request merged**